### DISCUSSION TITLE
Separate registries for pushing and pulling stages

### DISCUSSION THREAD

**Discussion by Zsmith32** [werf/werf#4081] [General]
In our environment we have different registries for our containers. We have one which we are allowed to push/pull from and the others are pull only registries.

At the moment we have our WERF_REPO set to the one we can push to to store our stages. But we'd like to be able to have werf pull from the other read only registries as they are closer to our ci environment and we pull far more often then push.

Is this possible currently or would this need to be a new feature?

Edit: forgot to mention the images are replicated almost instantly from the push/pull repos to the pull only repos. I think we do this for governance reasons.

**Comment by Zsmith32** [comment]
I've read the page on storage layouts. Seems close to what we need, but our caching registry is readonly so I don't think it will be able to serve as the werf caching repository.

**Reply by ilya-lesikov** [reply]
Will your read-only registry be able to remove images now missing (due to `werf cleanup`) from the primary registry?

**Reply by Zsmith32** [reply]
Deletes on the primary are automatically replicated to the read-only the same as pushes.

**Comment by distorhead** [comment]
Seems you need cache-repo function, but in read-only mode. All of cache-repos will be populated and cleaned outside of werf by replication.  We would be happy to see PR, which adds `--cache-repo-read-only=true` mode for example. In this mode werf simply should not push images into each cache-repo, only pull.
